autoprecompiles/src/memory_optimizer.rs

```diff
@@ -16,10 +16,19 @@ pub fn optimize_memory<
     V: Hash + Eq + Clone + Ord + Display,
     M: MemoryBusInteraction<T, V>,
 >(
-    mut system: ConstraintSystem<T, V>,
+    system: ConstraintSystem<T, V>,
     solver: &mut impl Solver<T, V>,
-    memory_bus_id: u64,
+    memory_bus_id: Option<u64>,
 ) -> ConstraintSystem<T, V> {
+    let memory_bus_id = match memory_bus_id {
+        Some(id) => id,
+        None => {
+            return system;
+        }
+    };
+
+    let mut system = system.clone();
```

Comment:
@chriseth I think I took this from the original code, but I can't really understand why we need a copy here

Reply:
So I don't think we need a copy here.

Reply:
thanks